pkg/controller/core.oam.dev/v1alpha2/application/application_controller.go

```diff
@@ -292,34 +292,34 @@ func (r *Reconciler) handleFinalizers(ctx monitorContext.Context, app *v1beta1.A
 		if !meta.FinalizerExists(app, resourceTrackerFinalizer) {
 			meta.AddFinalizer(app, resourceTrackerFinalizer)
 			ctx.Info("Register new finalizer for application", "finalizer", resourceTrackerFinalizer)
-			return true, ctrl.Result{}, errors.Wrap(r.Client.Update(ctx, app), errUpdateApplicationFinalizer)
+			return true, ctrl.Result{RequeueAfter: common2.ApplicationReSyncPeriod}, errors.Wrap(r.Client.Update(ctx, app), errUpdateApplicationFinalizer)
 		}
 	} else {
 		if meta.FinalizerExists(app, resourceTrackerFinalizer) {
 			rootRT, currentRT, historyRTs, cvRT, err := resourcetracker.ListApplicationResourceTrackers(ctx, r.Client, app)
 			if err != nil {
-				return true, ctrl.Result{}, err
+				return true, ctrl.Result{RequeueAfter: common2.ApplicationReSyncPeriod}, err
 			}
 			result, err := r.gcResourceTrackers(ctx, handler, common.ApplicationDeleting, true)
 			if err != nil {
 				return true, result, err
 			}
 			if rootRT == nil && currentRT == nil && len(historyRTs) == 0 && cvRT == nil {
 				meta.RemoveFinalizer(app, resourceTrackerFinalizer)
-				return true, ctrl.Result{}, errors.Wrap(r.Client.Update(ctx, app), errUpdateApplicationFinalizer)
+				return true, ctrl.Result{RequeueAfter: common2.ApplicationReSyncPeriod}, errors.Wrap(r.Client.Update(ctx, app), errUpdateApplicationFinalizer)
 			}
 			return true, result, err
 		}
 	}
-	return false, ctrl.Result{}, nil
+	return false, ctrl.Result{RequeueAfter: common2.ApplicationReSyncPeriod}, nil
 }
 
 func (r *Reconciler) endWithNegativeCondition(ctx context.Context, app *v1beta1.Application, condition condition.Condition, phase common.ApplicationPhase) (ctrl.Result, error) {
 	app.SetConditions(condition)
 	if err := r.patchStatus(ctx, app, phase); err != nil {
-		return ctrl.Result{}, errors.WithMessage(err, "cannot update application status")
+		return ctrl.Result{RequeueAfter: common2.ApplicationReSyncPeriod}, errors.WithMessage(err, "cannot update application status")
 	}
-	return ctrl.Result{}, fmt.Errorf("object level reconcile error, type: %q, msg: %q", string(condition.Type), condition.Message)
+	return ctrl.Result{RequeueAfter: common2.ApplicationReSyncPeriod}, fmt.Errorf("object level reconcile error, type: %q, msg: %q", string(condition.Type), condition.Message)
```

Comment:
we shouldn't return both requeue and erorr. That'll cause redundant reconcile.

Reply:
Yes. It's a mistake.